### PR TITLE
fix: improved symlink guards for sandbox, runfiles & execroot

### PR DESCRIPTION
**Pull request by gregmagolan**
Fixes https://github.com/aspect-build/rules_js/issues/220

Symlink guards improved & new unit test cases added

### CODE REVIEW

File: examples/webpack_cli/index.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+const {
```
Comment (alexeagle) [inline examples/webpack_cli/index.js:1]: where did the example come from? maybe attribution in the README

File: npm/private/npm_translate_lock.bzl [hunk 2]
```diff
@@ -598,9 +597,11 @@ load("@aspect_rules_js//npm/private:npm_linked_packages.bzl", "npm_linked_packag
                     }
 
     if fp_links:
-        defs_bzl_header.append("""load("@aspect_rules_js//npm/private:npm_link_package.bzl",
+        defs_bzl_header.append("""load(
+    "@aspect_rules_js//npm/private:npm_link_package.bzl",
```
Comment (alexeagle) [inline npm/private/npm_translate_lock.bzl:601]: nit: unrelated refactoring?

File: js/private/node-patches/src/fs.ts [hunk 3]
```diff
@@ -549,73 +454,270 @@ export const patcher = (fs: any = _fs, roots: string[]) => {
...
+                    const stat = fs.statSync(origRealpathSync(f))
                     patchDirent(v, stat)
                 }
             }
         }
     }
+
+    function nextHop(loc: string, cb: (next: string | false) => void): void {
+        let nested = []
+        const oneHop = (maybe, cb: (next: string | false) => void) => {
```
Comment (alexeagle) [inline js/private/node-patches/src/fs.ts:496]: this `cb` is shadowing the one from the outer function, maybe we should have a lint for that?

**Review by alexeagle — APPROVED** [review]
wow lots of new code, too much to review line-by-line in detail!

File: npm/private/npm_translate_lock.bzl (hunk 2 above)
Comment (gregmagolan) [reply npm/private/npm_translate_lock.bzl:620]: to satisfy lint; not sure how it got out of sync; but I'll split out the example into its own commit with proper attribution and then this change will go with that.

File: examples/webpack_cli/index.js (hunk 1 above)
Comment (gregmagolan) [reply examples/webpack_cli/index.js:1]: 👍

File: js/private/node-patches/src/fs.ts (hunk 3 above)
Comment (gregmagolan) [reply js/private/node-patches/src/fs.ts:496]: in this case it is intentional since it is the same cb.
```
const oneHop = (maybe, cb: (next: string | false) => void) => {...}
```
is followed by
```
oneHop(loc, cb)
```
which just passed the `cb` passed to the function forward
